src/Components/Authentication/helpers.ts

```diff
@@ -0,0 +1,36 @@
+export const handleSubmit = (
+  url: string,
+  csrf: string,
+  redirectUrl?: string
+) => async (values, formikBag) => {
+  try {
+    const data = await sendAuthData(url, { _csrf: csrf, ...values })
+    if (data.success) {
+      if (redirectUrl) {
+        document.location.pathname = redirectUrl
+      }
+    } else {
+      formikBag.setStatus(data)
+    }
+  } catch (err) {
+    formikBag.setStatus(err)
+  }
+}
+
+export async function sendAuthData(
+  url: string,
+  values: { [key: string]: any; _csrf: string }
+) {
+  const res = await fetch(url, {
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+      "X-Requested-With": "XMLHttpRequest",
+    },
+    method: "POST",
+    credentials: "same-origin",
+    body: JSON.stringify(values),
+  })
+  const data = await res.json()
+  return data
+}
```

Comment:
We'll want to make sure all of the same steps for signup/register in https://github.com/artsy/force/blob/master/src/desktop/models/logged_out_user.coffee are covered here too -- maybe this seems a little backwards, but have we considered passing up the Formik data up to Force and simply reuse the Backbone logic we already have for login/signup?